app/models/miq_alert_status.rb

```diff
@@ -1,4 +1,21 @@
+require 'ancestry'
+
 class MiqAlertStatus < ApplicationRecord
+  SEVERITY_LEVELS = %w(error warning info).freeze
+
+  has_ancestry
   belongs_to :miq_alert
   belongs_to :resource, :polymorphic => true
+  has_many :miq_alert_status_actions, -> { order "created_at" }, :dependent => :destroy
+  virtual_column :assignee, :type => :string
```

Comment:
@moolitayer In a follow up PR you can optimize this with a `:uses => :miq_alert_status_actions` The :uses clause lets the reporting engine know to preload that.  Alternately there might be a pure SQL way to do this, which perhaps @kbrock can help you with via the `:arel =>` option.